csharp/src/Ice/SslTransceiver.cs

```diff
@@ -446,203 +445,117 @@ private void ReadCompleted(IAsyncResult result)
         private bool RemoteCertificateValidationCallback(
             object sender,
             X509Certificate certificate,
-            X509Chain chainEngine,
-            SslPolicyErrors policyErrors)
+            X509Chain chain,
+            SslPolicyErrors errors)
         {
-            var chain = new X509Chain(_engine.UseMachineContext);
-            try
-            {
-                if (_engine.CheckCRL == 0)
-                {
-                    chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
-                }
+            var message = new StringBuilder();
 
-                X509Certificate2Collection? caCerts = _engine.CaCerts;
-                if (caCerts != null)
-                {
-                    //
-                    // We need to set this flag to be able to use a certificate authority from the extra store.
-                    //
-                    chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
-                    foreach (X509Certificate2 cert in caCerts)
-                    {
-                        chain.ChainPolicy.ExtraStore.Add(cert);
-                    }
-                }
-
-                string message = "";
-                int errors = (int)policyErrors;
-                if (certificate != null)
+            if ((errors & SslPolicyErrors.RemoteCertificateNotAvailable) > 0)
+            {
+                // The RemoteCertificateNotAvailable case is not possible for an outgoing connection. Since .NET
+                // requires an authenticated connection, the remote peer closes the socket if it does not have a
+                // certificate to provide.
+                if (_incoming)
```

Comment:
Should we assert for the client case if we are sure it's not possible and always return false instead of allowing the connection?

Reply:
We could change this if to `if(!_incoming || _engine.TlsServerOptions.RequireClientCertificate)`?

Reply:
sure